### ISSUE TITLE
Consider creating testkit module for helper classes which are reused across tests

### ISSUE THREAD

**Issue by vlsi** [assertj/assertj#2856]
#### Feature summary

The current test code relies on classes which can't be shared across modules.
For instance: `ConcreteAssert`, `AssertionsUtil`

It makes it harder to reuse/copy existing tests into another modules.
For instance, in https://github.com/assertj/assertj/pull/2855 I add new Kotlin-based tests, and I can't reuse the existing `ConcreteAssert`, `AssertionsUtil`.

It would be nice if `src/test` code included only classes with `@Test` methods, and the "test framework" should better be located in its own `src/main` so it can be reused as a dependency.

**Comment by scordio** [comment]
Hi @vlsi, thanks for the suggestion! Now that also `assertj-guava` is part of the codebase, we have more boilerplate in the testing area so a testkit module might make sense as long as we don't publish it to the Nexus.

**Comment by vlsi** [comment]
Of course, testkit can be unpublished, however, I see nothing wrong with publishing it.
At the end of the day, you might expect that people would create their own AssertJ extensions, and they would need to test them. So testkit would be more than useful for them.
